trieTools/zeroBalanceSystemAccountChecker/main.go

```diff
@@ -0,0 +1,409 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go-core/core/pubkeyConverter"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-tools-go/elasticreindexer/config"
+	"github.com/ElrondNetwork/elrond-tools-go/elasticreindexer/elastic"
+	"github.com/ElrondNetwork/elrond-tools-go/trieTools/trieToolsCommon"
+	sysAccConfig "github.com/ElrondNetwork/elrond-tools-go/trieTools/zeroBalanceSystemAccountChecker/config"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/pelletier/go-toml"
+	"github.com/urfave/cli"
+)
+
+const (
+	logFilePrefix   = "system-account-zero-tokens-balance-checker"
+	addressLength   = 32
+	outputFilePerms = 0644
+	tomlFile        = "./config.toml"
+)
+
+type crossTokenChecker interface {
+	crossCheckExtraTokens(tokens map[string]struct{}) ([]string, error)
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "Tokens exporter CLI app"
+	app.Usage = "This is the entry point for the tool that checks which tokens are not used anymore(only stored in system account)"
+	app.Flags = getFlags()
+	app.Authors = []cli.Author{
+		{
+			Name:  "The Elrond Team",
+			Email: "contact@elrond.com",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+		return startProcess(c)
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Error(err.Error())
+		os.Exit(1)
+		return
+	}
+}
+
+func startProcess(c *cli.Context) error {
+	flagsConfig := getFlagsConfig(c)
+
+	_, errLogger := trieToolsCommon.AttachFileLogger(log, logFilePrefix, flagsConfig.ContextFlagsConfig)
+	if errLogger != nil {
+		return errLogger
+	}
+
+	err := logger.SetLogLevel(flagsConfig.LogLevel)
+	if err != nil {
+		return err
+	}
+
+	globalAddressTokensMap, shardAddressTokensMap, err := readInputs(flagsConfig.TokensDirectory)
+	if err != nil {
+		return err
+	}
+
+	globalExtraTokens, extraTokensPerShard, err := exportSystemAccZeroTokensBalances(globalAddressTokensMap, shardAddressTokensMap)
+	if err != nil {
+		return err
+	}
+
+	if flagsConfig.CrossCheck {
+		err = crossCheckExtraTokens(globalExtraTokens, extraTokensPerShard)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = saveResult(extraTokensPerShard, flagsConfig.Outfile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func readInputs(tokensDir string) (map[string]map[string]struct{}, map[uint32]map[string]map[string]struct{}, error) {
```

Comment:
this incredible hard to comprehend why you needed `map[uint32]map[string]map[string]struct{}`
why not use proper (even nested) structs?

Reply:
Replaced `map[string]map[string]struct{}` with new interface called `AddressTokensMap`